src/Symfony/Bundle/FrameworkBundle/Controller/AbstractController.php

```diff
@@ -64,7 +65,8 @@ public function setContainer(ContainerInterface $container)
     protected function getParameter(string $name)
     {
         if (!$this->container->has('parameter_bag')) {
-            throw new \LogicException('The "parameter_bag" service is not available. Try running "composer require dependency-injection:^4.1"');
+            throw new ServiceNotFoundException('parameter_bag', null, null, array(),
+                sprintf('The "getParameter()" method of the controller can\'t get the value of the "%s" parameter because the "parameter_bag" service could not be located. Either enable autoconfigure for the controller service or inject a ServiceLocator referencing a "parameter_bag" service manually in the controller.', $name));
```

Comment:
You missed my last comment here it is :) Also this should really be on one line, not two. 

`sprintf('The "%s::getParameter()" method is missing a parameter bag to work properly. Did you forget to register your controller as a service subscriber? This can be fixed e.g. by using autoconfiguration or by manually wiring a "parameter_bag" in the service locator passed to the controller.', get_class($this))`

(the name of the parameter is of no importance IMHO, but the controller class IS.)

Reply:
Thought you meant only the string with the previous one-line remark. Is the Symfony code style on layout stuff like that documented somewhere? It's confusing that most projects (including our internal ones) explicitly prohibit lines beyond common-screen limits (ie. ~120 characters) and Symfony 'prefers oneliners'.